src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -261,31 +227,25 @@ def to_dict(self) -> Dict[str, Any]:
         if self._realization_states:
             _dict["reals"] = self._realization_states
 
-        for step_index_tuple, step_state in self._step_states.items():
-            real_id = step_index_tuple[0]
-            step_id = step_index_tuple[1]
+        for real_id, step_state in self._step_states.items():
             if "reals" not in _dict:
                 _dict["reals"] = {real_id: {}}
-            if "steps" not in _dict["reals"][real_id]:
-                _dict["reals"][real_id]["steps"] = {}
-            if step_id not in _dict["reals"][real_id]["steps"]:
-                _dict["reals"][real_id]["steps"][step_id] = step_state
-                _dict["reals"][real_id]["steps"][step_id]["jobs"] = {}
+            if "step" not in _dict["reals"][real_id]:
+                _dict["reals"][real_id]["step"] = step_state
 
         for job_tuple, job_values_dict in self._job_states.items():
             real_id = job_tuple[0]
-            step_id = job_tuple[1]
             if "reals" not in _dict:
                 _dict["reals"] = {}
             if real_id not in _dict["reals"]:
                 _dict["reals"][real_id] = {}
-            if "steps" not in _dict["reals"][real_id]:
-                _dict["reals"][real_id]["steps"] = {}
-            if step_id not in _dict["reals"][real_id]["steps"]:
-                _dict["reals"][real_id]["steps"][step_id] = {"jobs": {}}
+            if "step" not in _dict["reals"][real_id]:
```

Comment:
All the default dicts :) chatgpt knows how to approach this:
```python
class DynamicDefaultDict(dict):
    def __missing__(self, key):
        if key == 'fruit':
            value = defaultdict(int)
        elif key == 'vegetable':
            value = defaultdict(lambda: defaultdict(int))
        else:
            value = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
        
        self[key] = value
        return value

# Create an instance of the custom dictionary
d = DynamicDefaultDict()

# Use it like a defaultdict
d['fruit']['apple'] += 1
print(d['fruit']['apple'])  # Output: 1

d['vegetable']['carrot']['orange'] += 1
print(d['vegetable']['carrot']['orange'])  # Output: 1

d['other']['key1']['key2']['key3'] += 1
print(d['other']['key1']['key2']['key3'])  # Output: 1
```

Reply:
That should be in a different PR :)

Reply:
Agree :)